src/Migrations/M032MigrateCampaignCoversToP4ColumnsBlock.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+declare(strict_types=1);
+
+namespace P4\MasterTheme\Migrations;
+
+use P4\MasterTheme\MigrationRecord;
+use P4\MasterTheme\MigrationScript;
+use P4\MasterTheme\BlockReportSearch\BlockSearch;
+use WP_Block_Parser;
+
+/**
+ * Migrate Campaign covers block to Planet4 columns block.
+ */
+class M032MigrateCampaignCoversToP4ColumnsBlock extends MigrationScript
+{
+    private const BLOCK_NAME = 'planet4-blocks/covers';
+    private const POST_TYPES = [ 'page', 'post', 'action', 'campaign' ];
+    private const COVER_TYPE = 'campaign';
+    private const OLD_COVER_TYPES = [
+        '1' => 'take-action',
+        '2' => 'campaign',
+        '3' => 'content',
+    ];
+
+    /**
+     * Extract campaign covers block from page/posts and transform it into Planet4 columns block.
+     *
+     * @param MigrationRecord $record Information on the execution, can be used to add logs.
+     * phpcs:disable SlevomatCodingStandard.Functions.UnusedParameter -- interface implementation
+     */
+    public static function execute(MigrationRecord $record): void
+    {
+        $parser = new WP_Block_Parser();
```

Comment:
Maybe this line could be moved above line 44 to only create a new instance of the `WP_Block_Parser`  if posts are using the Campaign Covers block.